src/framework/linalg/matrix_utils/matrix_defs.hpp

```diff
@@ -195,14 +196,20 @@ const cmatrix_t Matrix::SWAP =
                                    {{0, 0}, {1, 0}, {0, 0}, {0, 0}},
                                    {{0, 0}, {0, 0}, {0, 0}, {1, 0}}});
 
+const cmatrix_t Matrix::ECR =
+    Utils::make_matrix<complex_t>({{{0, 0}, {0, 0}, {1. / std::sqrt(2.), 0}, {0, 1. / std::sqrt(2.)}},
+                                   {{0, 0}, {0, 0}, {0, 1 / std::sqrt(2.)}, {1. / std::sqrt(2.), 0}},
+                                   {{1. / sqrt(2.), 0}, {0, -1. / std::sqrt(2.)}, {0, 0}, {0, 0}},
+                                   {{0, -1. / std::sqrt(2.)}, {1. / std::sqrt(2.), 0}, {0, 0}, {0, 0}}});
+
```

Comment:
I think you've got an endianness difference in this matrix definition compared to Terra.

Reply:
I just clicked through to Terra's documentation - they look to be flipped compared to how we actually implement the gate definition in code.

Reply:
Sigh, I just copied it from the docs but I copied the matrix. I'll fix this in the morning.

Reply:
Wait no, the docs are kind of right, they're just phrased a bit confusingly.  The top matrix is what Qiskit actually produces - the bottom one is what it would look like in big-endian.

Reply:
ECR gate looks wrong in qiskit. It should implement an RZX(pi/2) matrix, the echoing part only matters if you are decomposing.